### PR TITLE
git_resource: improvements to url parsing

### PR DESCRIPTION
**Pull request by nicks**
Hello @kogi, @maiamcc,

Please review the following commits I made in branch nicks/issue135:

ac4d02e9fba259582b8e3291fcb8cf83969c61d4 (2021-03-03 15:58:56 -0500)
git_resource: improvements to url parsing
- handle URLs with the format http://user@host/
- handle repositories where "master" is not the default repo branch

fixes https://github.com/tilt-dev/tilt-extensions/issues/135

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: git_resource/Tiltfile [hunk 1]
```diff
@@ -106,11 +122,16 @@ def _parse_repository_url(url):
...
-        tree = '#master'
+        tree = ''
     else:
         parts = url.split(splitter)
         tree = splitter + parts[-1]  # last item
-        url = splitter.join(parts[:-1])  # everything else
+
+        # Make sure we didn't peel of a scheme://user@host/ url
+        if splitter == '@' and '/' in tree:
+            tree = ''
```
Comment (KOGI) [inline git_resource/Tiltfile:132]: this doesn't seem to handle the case of `scheme://user@host/some/path@commit`

File: git_resource/test/Tiltfile [hunk 2]
```diff
@@ -1,6 +1,34 @@
...
+   ("git@example.com/path/to/repo.git", "repo", "")),
+  ('git@example.com/path/to/repo.git@myRevisionSha',
+   ("git@example.com/path/to/repo.git", "repo", "@myRevisionSha")),
+  ('git@example.com/path/to/repo.git#myBranchNameHere',
+   ("git@example.com/path/to/repo.git", "repo", "#myBranchNameHere")),
+  ('https://user@example.com/path/repo.git',
+   ('https://user@example.com/path/repo.git', 'repo', '')),
+  ('ssh://user@example.com:54321/path/repo.git',
+   ('ssh://user@example.com:54321/path/repo.git', 'repo', '')),
+  ('https://user@example.com/path/repo.git@myRevisionSha',
```
Comment (KOGI) [inline git_resource/test/Tiltfile:21]: then again, maybe it does...

File: git_resource/test/Tiltfile [hunk 3]
```diff
@@ -1,6 +1,34 @@
 os.putenv('TILT_GIT_RESOURCE_CHECKOUT_DIR', os.path.abspath('./.git-sources'))
 load('../Tiltfile', 'git_checkout', 'deploy_from_dir')
 
+symbols = load_dynamic('../Tiltfile')
+
+url_test_cases = [
```
Comment (KOGI) [inline git_resource/test/Tiltfile:6]: I'm sure it matches Tilt's coding standard, but the forced-line-wrapping here makes this super difficult to read

Comment (KOGI) [reply git_resource/test/Tiltfile:6]: Food for thought:
```python
url_test_cases = [
  ('https://github.com/tilt-dev/tilt-example-html',        ('https://github.com/tilt-dev/tilt-example-html', 'tilt-example-html', ''                 )),
  ('git@example.com/path/to/repo',                         ('git@example.com/path/to/repo',                  'repo',              ''                 )),
  ('git@example.com/path/to/repo.git',                     ('git@example.com/path/to/repo.git',              'repo',              ''                 )),
  ('git@example.com/path/to/repo.git@myRevisionSha',       ('git@example.com/path/to/repo.git',              'repo',              '@myRevisionSha'   )),
  ('git@example.com/path/to/repo.git#myBranchNameHere',    ('git@example.com/path/to/repo.git',              'repo',              '#myBranchNameHere')),
  ('https://user@example.com/path/repo.git',               ('https://user@example.com/path/repo.git',        'repo',              ''                 )),
  ('ssh://user@example.com:54321/path/repo.git',           ('ssh://user@example.com:54321/path/repo.git',    'repo',              ''                 )),
  ('https://user@example.com/path/repo.git@myRevisionSha', ('https://user@example.com/path/repo.git',        'repo',              '@myRevisionSha'   )),
]
```

**Comment by KOGI** [comment]
Left a few thoughts, but otherwise LGTM :)

File: git_resource/test/Tiltfile (hunk 3 above)
Comment (nicks) [reply git_resource/test/Tiltfile:6]: ya, totally agree, i'm going to create a helper function to make this more readable

File: git_resource/test/Tiltfile (hunk 2 above)
Comment (nicks) [reply git_resource/test/Tiltfile:21]: : )

File: git_resource/Tiltfile (hunk 1 above)
Comment (nicks) [reply git_resource/Tiltfile:132]: it's subtle! ya, this only gets the last part

**Comment by nicks** [comment]
thanks @KOGI ! going to merge in a bit, but lemme know if this breaks anything :grimacing: ; we really need a better way to test these changes against downstream projects

**Pull request merged**